Products/CMFPlone/patterns/__init__.py

```diff
@@ -242,6 +241,8 @@ def tinymce(self):
 
         image_types = settings.image_objects or []
         folder_types = settings.contains_objects or []
+
+        site_path = generator.portal_url.replace(self.request['SERVER_URL'], '')
```

Comment:
ha! urlparse. nice.
but what do you mean with write-only code?
i have no probs with regex except that they are hard to understand if not clearly documented.